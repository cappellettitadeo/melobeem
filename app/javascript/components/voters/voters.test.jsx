import { useRequest } from "@umijs/hooks";
import { mount } from "enzyme";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Voters from "./voters";
const mockProps = {
  match: {
    params: `111`,
  },
};

jest.mock("@umijs/hooks");

describe("Components", () => {
  describe("Voters", () => {
    it("should render Vote component", () => {
      const component = renderer
        .create(
          <Router>
            <Voters id={mockProps} />
          </Router>
        )
        .toJSON();

      expect(component).toMatchSnapshot();
    });

    it("should render Loading state", () => {
      beforeEach(() => useRequest.mockReturnValue({ loading: true }));
      const component = mount(
        <Router>
          <Voters id={mockProps} />
        </Router>
      );

      // const loading = component.find(".loading").at(0);
      expect(component.exists(".loading")).toBeTruthy();
    });
  });
});
