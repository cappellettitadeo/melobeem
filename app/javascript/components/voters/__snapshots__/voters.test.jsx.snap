// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Voters should render Vote component 1`] = `
<div
  className="animation animation--loader"
>
  <div
    style={
      Object {
        "height": 120,
        "width": 120,
      }
    }
  />
</div>
`;
